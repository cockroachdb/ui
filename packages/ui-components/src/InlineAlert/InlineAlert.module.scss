@import "../styles/tokens.scss";

.root {
  display: flex;
  flex-direction: row;
  flex-grow: 1;
  border-left: solid crl-gutters(.5);
  border-radius: crl-gutters(.5);
  padding: crl-gutters(1.5) crl-gutters(3) crl-gutters(1.5) 0;
}

.container {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}

.title {
  line-height: crl-gutters(3);
  margin-bottom: crl-gutters(.5);
  font-weight: 600;
}

.icon-container {
  padding: 0 crl-gutters(2);
}

.intent-info {
  border-left-color: $color-background-info;
  background-color: $color-background-badge-info;
}

.intent-error {
  border-left-color: $color-background-danger;
  background-color: $color-background-badge-danger;
}

.intent-warning {
  border-left-color: $color-base-orange;
  background-color: $color-background-badge-warning;
}

.intent-success {
  border-left-color: $color-core-green-3;
  background-color: $color-core-green-1;
}
