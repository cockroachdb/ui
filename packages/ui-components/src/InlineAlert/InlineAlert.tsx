import React, { useMemo } from "react";
import classNames from "classnames/bind";

import styles from "./InlineAlert.module.scss";
import { Icon, IconTint } from "../Icon/Icon";

export type InlineAlertIntent = "info" | "error" | "success" | "warning";

const cx = classNames.bind(styles);

interface InlineAlertBaseProps {
  intent?: InlineAlertIntent;
  className?: string;
}

interface InlineAlertWithTitleProps {
  title: React.ReactNode;
  description?: React.ReactNode;
}

interface InlineAlertWithDescriptionProps {
  title?: React.ReactNode;
  description: React.ReactNode;
}

export type InlineAlertProps = InlineAlertBaseProps &
  (InlineAlertWithTitleProps | InlineAlertWithDescriptionProps);

export const InlineAlert: React.FC<InlineAlertProps> = ({
  title,
  description,
  intent = "info",
  className,
}) => {
  const iconName = useMemo(() => {
    switch (intent) {
      case "error":
        return "ErrorCircleFilled";
      case "success":
        return "CheckCircleFilled";
      case "warning":
        return "Caution";
      case "info":
      default:
        return "InfoCircleFilled";
    }
  }, [intent]);

  const iconTint = useMemo<IconTint>(() => {
    switch (intent) {
      case "error":
        return "red";
      case "success":
        return "green";
      case "warning":
        return "orange";
      case "info":
      default:
        return "blue";
    }
  }, [intent]);

  return (
    <div className={cx("root", `intent-${intent}`, className)}>
      <div className={cx("icon-container")}>
        <Icon iconName={iconName} size="default" tint={iconTint} />
      </div>
      <div className={cx("container")}>
        {title && <div className={cx("title")}>{title}</div>}
        {description && <div className={cx("description")}>{description}</div>}
      </div>
    </div>
  );
};
