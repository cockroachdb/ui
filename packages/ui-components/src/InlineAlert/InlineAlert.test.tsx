import React from "react";
import { shallow } from "enzyme";
import { InlineAlert, InlineAlertIntent } from "./InlineAlert";

const alertIntents: InlineAlertIntent[] = [
  undefined, // should be used default intent
  "warning",
  "success",
  "error",
  "info",
];

describe("InlineAlert", () => {
  alertIntents.forEach(intent => {
    test(`renders with ${intent} intent`, () => {
      const wrapper = shallow(
        <InlineAlert intent={intent} title="Hello world!" />,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("renders with provided title", () => {
    const title = "Hello world!";
    const wrapper = shallow(<InlineAlert title={title} />);
    expect(wrapper.text()).toContain(title);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with provided description", () => {
    const description = "Hello world!";
    const wrapper = shallow(<InlineAlert description={description} />);
    expect(wrapper.text()).toContain(description);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with provided title and description", () => {
    const title = "Hello world!";
    const description =
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
    const wrapper = shallow(
      <InlineAlert title={title} description={description} />,
    );
    expect(wrapper.text()).toContain(title);
    expect(wrapper.text()).toContain(description);
    expect(wrapper).toMatchSnapshot();
  });
});
