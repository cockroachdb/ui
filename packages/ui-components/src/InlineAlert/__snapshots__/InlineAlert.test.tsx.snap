// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineAlert renders with error intent 1`] = `
<div
  className="root intent-error"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="ErrorCircleFilled"
      size="default"
      tint="red"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with info intent 1`] = `
<div
  className="root intent-info"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="InfoCircleFilled"
      size="default"
      tint="blue"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with provided description 1`] = `
<div
  className="root intent-info"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="InfoCircleFilled"
      size="default"
      tint="blue"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="description"
    >
      Hello world!
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with provided title 1`] = `
<div
  className="root intent-info"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="InfoCircleFilled"
      size="default"
      tint="blue"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with provided title and description 1`] = `
<div
  className="root intent-info"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="InfoCircleFilled"
      size="default"
      tint="blue"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
    <div
      className="description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with success intent 1`] = `
<div
  className="root intent-success"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="CheckCircleFilled"
      size="default"
      tint="green"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with undefined intent 1`] = `
<div
  className="root intent-info"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="InfoCircleFilled"
      size="default"
      tint="blue"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
  </div>
</div>
`;

exports[`InlineAlert renders with warning intent 1`] = `
<div
  className="root intent-warning"
>
  <div
    className="icon-container"
  >
    <Component
      iconName="Caution"
      size="default"
      tint="orange"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Hello world!
    </div>
  </div>
</div>
`;
